pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -102,8 +104,12 @@ if [[ "$CV_MODE" == "lvm" ]]; then
 	# remove the temporary directory
 	rm --recursive --force "$TMP_DIR"
 else
+	ARGS=(--device ${DEVICE} --no-systemd --no-tmpfs)
+	if [ -n "$METADATA_DEVICE" ]; then
+		ARGS+=(--block.db ${METADATA_DEVICE})
+	fi
 	# ceph-volume raw mode only supports bluestore so we don't need to pass a store flag
-	ceph-volume "$CV_MODE" activate --device "$DEVICE" --no-systemd --no-tmpfs
+	ceph-volume "$CV_MODE" activate "${ARGS[@]}"
```

Comment:
This only applies to LVM-based OSDs, right? Were OSDs not working when on non-PVC metadata devices? Or why is this change needed?

Reply:
No this applies to raw mode, look at the condition.

Reply:
removed.